libdnf/rpm/package_sack.cpp

```diff
@@ -187,6 +188,14 @@ void PackageSack::Impl::setup_excludes_includes(bool only_main) {
     }
 }
 
+void PackageSack::Impl::add_excludes(const PackageSet & excludes) {
+    if (&(pkg_excludes.get()->get_map()) == nullptr) {
```

Comment:
`if (!pkg_excludes)` Is may be what you are looking for. What about to modify it to:
```
if (!pkg_excludes) {
    pkg_excludes.reset(new libdnf::solv::SolvMap(*excludes.p_impl));
} else {
    *pkg_excludes |= *excludes.p_impl;
}
considered_uptodate = false;
```